src/components/ArtifactTimelineItemContent.vue

```diff
@@ -0,0 +1,180 @@
+<template>
+  <p-card
+    class="artifact-timeline-item-content"
+    :class="classes.root"
+    tabindex="0"
+    :flat="!expanded"
+    @keyup.self.enter="toggleExpanded"
+  >
+    <div class="artifact-timeline-item-content__heading" :class="classes.heading" @click="toggleExpanded">
+      <p-heading heading="6">
+        <p-link :to="routes.artifact(artifact.id)" :title="artifact.id" @click.stop>
+          {{ shortId }}
+        </p-link>
+      </p-heading>
+
+      <template v-if="artifact.flowRunId">
+        <div class="artifact-timeline-item-content__icon-text-container">
+          {{ localization.info.flowRun }}
+          <FlowRunIconText class="artifact-timeline-item-content__icon-text" :flow-run-id="artifact.flowRunId" />
+        </div>
+      </template>
+
+      <template v-if="artifact.taskRunId">
+        <div class="artifact-timeline-item-content__icon-text-container">
+          {{ localization.info.taskRun }}
+          <TaskRunIconText class="artifact-timeline-item-content__icon-text" :task-run-id="artifact.taskRunId" />
+        </div>
+      </template>
+    </div>
+
+    <template v-if="expanded">
+      <p-divider />
+
+      <template v-if="expandedDebounced">
+        <section class="artifact-timeline-item-content__body">
+          <template v-if="artifact.description">
+            <p-markdown-renderer :text="artifact.description" />
+
+            <p-divider />
+          </template>
+
+          <ArtifactDataView :artifact="artifact" />
+        </section>
+      </template>
+
+      <template v-else>
+        <div class="artifact-timeline-item-content__loading">
+          <p-loading-icon />
+        </div>
+      </template>
+    </template>
+  </p-card>
+</template>
+
+<script lang="ts" setup>
+  import { useDebouncedRef } from '@prefecthq/vue-compositions'
+  import { isArray } from 'lodash'
+  import { computed } from 'vue'
+  import ArtifactDataView from '@/components/ArtifactDataView.vue'
+  import FlowRunIconText from '@/components/FlowRunIconText.vue'
+  import TaskRunIconText from '@/components/TaskRunIconText.vue'
+  import { useWorkspaceRoutes } from '@/compositions'
+  import { localization } from '@/localization'
+  import { Artifact } from '@/models'
+  import { isNullish } from '@/utilities'
+
+  type Expanded = boolean | unknown[] | undefined | null
+
+  const props = defineProps<{
+    artifact: Artifact,
+    // TODO: I don't think this logic makes sense here
+    // but I want to wait to create a more generic expanded card component
+    expanded?: Expanded,
```

Comment:
Mentioned this in other PR. But it doesn't seem like expanded needs to be a prop? There a lot of logic around this because its a v-model when I think this could be handled as a simple boolean ref internal to the component.